File: app/src/main/java/com/rupam/notes/Data/NoteAdapter.java
package com.rupam.notes.Data;

import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.rupam.notes.Activities.NoteDetailsActivity;
import com.rupam.notes.Model.Note;
import com.rupam.notes.R;

import java.util.Date;
import java.util.List;
import java.util.Random;

public class NoteAdapter extends RecyclerView.Adapter<NoteAdapter.ViewHolder> {

    private Context context;
    private List<Note> noteList;


    public NoteAdapter(Context context, List<Note> noteList) {
        this.context = context;
        this.noteList = noteList;
    }
    @NonNull
    @Override
    public NoteAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.note_row, parent, false);

        return new ViewHolder(view, context);
    }

    @Override
    public void onBindViewHolder(@NonNull NoteAdapter.ViewHolder holder, int position) {

        Note note = noteList.get(position);

        holder.title.setText(note.getTitle().toUpperCase());
        holder.body.setText(note.getNoteBody());

        java.text.DateFormat dateFormat = java.text.DateFormat.getDateInstance();
        String formattedDate = dateFormat.format(new Date(Long.valueOf(note.getDateAdded())).getTime());

        holder.dateAdded.setText(formattedDate);
    }

    @Override
    public int getItemCount() {
        return noteList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        public TextView title;
        public TextView body;
        public TextView dateAdded;
        private View cardView;
        String userId;

        public ViewHolder(@NonNull final View itemView, final Context ctx) {
            super(itemView);

            context = ctx;

            title = itemView.findViewById(R.id.titleNoteRow);
            body = itemView.findViewById(R.id.descNoteRow);
            dateAdded = itemView.findViewById(R.id.dateAdded);
            cardView = itemView.findViewById(R.id.cardViewRow);

            int[] androidColors = itemView.getResources().getIntArray(R.array.androidcolors);
            int randomAndroidColor = androidColors[new Random().nextInt(androidColors.length)];

            cardView.setBackgroundColor(randomAndroidColor);
           
            userId = null;
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {

                    Note selectedItem = noteList.get(getAdapterPosition());
                    String selectedKey = selectedItem.getKey();

//                    TODO: Goto detailsActivity with details
                    Intent intent = new Intent(context, NoteDetailsActivity.class);
                    intent.putExtra("title", title.getText().toString());
                    intent.putExtra("body", body.getText().toString());
                    intent.putExtra("userKey", selectedKey);  //Sends the selected key

                    ctx.startActivity(intent);
                }
            });
        }
    }
}

File: app/src/main/java/com/rupam/notes/Data/NoteAdapter.kt
package com.rupam.notes.Data

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.rupam.notes.Model.Note
import com.rupam.notes.R
import java.text.DateFormat
import java.util.*

class NoteAdapter(private val context: Context, private val noteList: MutableList<Note?>) : RecyclerView.Adapter<NoteAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.note_row, parent, false)
        return ViewHolder(view, context)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val note = noteList[position]
        holder.title.text = note?.title?.toUpperCase()
        holder.body.text = note?.noteBody
        val dateFormat = DateFormat.getDateInstance()
        val formattedDate = dateFormat.format(Date(java.lang.Long.valueOf(note?.dateAdded!!)).time)
        holder.dateAdded.text = formattedDate
    }

    override fun getItemCount(): Int {
        return noteList.size
    }

    inner class ViewHolder(itemView: View, var ctx: Context) : RecyclerView.ViewHolder(itemView) {
        var title: TextView
        var body: TextView
        var dateAdded: TextView
        private val cardView: View
        var userId: String?

        init {

            title = itemView.findViewById(R.id.titleNoteRow)
            body = itemView.findViewById(R.id.descNoteRow)
            dateAdded = itemView.findViewById(R.id.dateAdded)
            cardView = itemView.findViewById(R.id.cardViewRow)

            val androidColors = itemView.resources.getIntArray(R.array.androidcolors)
            val randomAndroidColor = androidColors[Random().nextInt(androidColors.size)]
            cardView.setBackgroundColor(randomAndroidColor)
            userId = null
            itemView.setOnClickListener {
                val selectedItem = noteList[adapterPosition]
                val selectedKey = selectedItem?.key

//                    TODO: Goto detailsActivity with details
////                val intent = Intent(context, NoteDetailsActivity::class.java)
//                intent.putExtra("title", title.text.toString())
//                intent.putExtra("body", body.text.toString())
//                intent.putExtra("userKey", selectedKey) //Sends the selected key
//                ctx.startActivity(intent)
            }
        }
    }

}